Dockerfile

```
@@ -0,0 +1,41 @@
+# Multi-stage build to download the just binary and create a minimal container
+FROM alpine:latest as downloader
+
+# Version will be passed from GitHub Actions during release builds
+ARG JUST_VERSION
+# Docker automatically provides these for multi-platform builds
+ARG TARGETARCH
+ARG TARGETVARIANT
+
+# Install tools needed to download and extract the binary
+RUN apk add --no-cache wget tar
+
+# Download the appropriate just binary for the target architecture
+# Map Docker's platform names to just's release naming convention
+RUN set -eux; \
+    case "$TARGETARCH" in \
+        "amd64") ARCH="x86_64-unknown-linux-musl" ;; \
+        "arm64") ARCH="aarch64-unknown-linux-musl" ;; \
+        "arm") \
+            # Handle ARM variants - v7 maps to armv7, others to generic arm
+            case "$TARGETVARIANT" in \
+                "v7") ARCH="armv7-unknown-linux-musleabihf" ;; \
+                *) ARCH="arm-unknown-linux-musleabihf" ;; \
+            esac ;; \
+        *) echo "Unsupported architecture: $TARGETARCH" && exit 1 ;; \
+    esac; \
+    wget -O just.tar.gz "https://github.com/casey/just/releases/download/${JUST_VERSION}/just-${JUST_VERSION}-${ARCH}.tar.gz"; \
+    tar -xzf just.tar.gz; \
+    chmod +x just
+
+# Use scratch for minimal final image - no OS, just the binary
+# This results in a ~10MB image vs ~50MB+ with a full OS
+FROM scratch
+COPY --from=downloader /just /usr/local/bin/just
+
+# Set working directory for when just is run
+WORKDIR /work
```

Perhaps something a little more appropriate if you're choosing a custom location?

```suggestion
WORKDIR /just
```

Is there a scenario where you would use this implicitly? (_no mounts to this location?_)

I suppose it provides a small convenience at runtime for those that are aware of it, otherwise it can be a tad annoying to inherit a WORKDIR from a base image when extending (_CoreDNS had this issue when they switched from Debian to Google Distroless as their base image which broke users with a confusing failure about their configs not being read as the change was not documented in any manner_).

If you do think this is worth having as a default, you should add a mention about this in the README docs you've added (_which only seem to touch on using the image as a distribution mechanism where the `WORKDIR` here is irrelevant in that `COPY` context_).

---

Feedback was ignored?